.github/workflows/_ci.yaml

```diff
@@ -399,6 +399,180 @@ jobs:
+        apiVersion: batch/v1
+        kind: Job
+        metadata:
+          name: ${{ github.run_id }}-${{ github.run_attempt }}-postprocess
+        spec:
+          template:
+            spec:
+              restartPolicy: Never
+              initContainers:
+              - name: download
+                image: amazon/aws-cli
+                command:
+                - aws
+                - s3
+                - cp
+                - --recursive
+                - --exclude
+                - "*"
+                - --include
+                - "${{ github.run_id }}-${{ github.run_attempt }}-rank*.zip"
+                - s3://jax-toolbox-eks-output/
+                - /opt/output
+                volumeMounts:
+                - mountPath: /opt/output
+                  name: output
+              containers:
+              - name: jax
+                image: ${{ needs.build-jax.outputs.DOCKER_TAG_FINAL }}
+                command:
+                - bash
+                - -exo
+                - pipefail
+                - -c
+                - nsys-jax-combine -o /opt/output/combined.zip /opt/output/*.zip --analysis communication
+                # FIXME: GPU not actually needed, but the test cluster doesn't have appropriate non-GPU nodes
+                resources:
+                  limits:
+                    nvidia.com/gpu: 1
+                volumeMounts:
+                - mountPath: /opt/output
+                  name: output
+              imagePullSecrets:
+              - name: ${{ github.run_id }}-${{ github.run_attempt }}-token
+              volumes:
+              - name: output
+                emptyDir: {}
+      JOB_DESCRIPTION: |
+        apiVersion: v1
+        kind: Service
+        metadata:
+          name: jax-headless-svc
+        spec:
+          clusterIP: None # clusterIP must be None to create a headless service
+          selector:
+            job-name: ${{ github.run_id }}-${{ github.run_attempt }}-jax # must match Job name
+        ---
+        apiVersion: batch/v1
+        kind: Job
+        metadata:
+          name: ${{ github.run_id }}-${{ github.run_attempt }}-jax
+        spec:
+          completions: 2 # number of nodes
+          parallelism: 2 # number of nodes
+          completionMode: Indexed
+          template:
+            spec:
+              subdomain: jax-headless-svc # has to match Service name
+              restartPolicy: Never
+              containers:
+              - name: jax
+                image: ${{ needs.build-jax.outputs.DOCKER_TAG_FINAL }}
+                ports:
+                - containerPort: 3389
+                command:
+                - sh
+                - -c
+                - |
+                  install-efa.sh
+                  nsys-jax \
+                    --output=/opt/output/rank%q{JOB_COMPLETION_INDEX}.zip \
+                    -- \
+                    jax-nccl-test \
+                    --coordinator-address \
```

Comment:
Why not use a [JobSet](https://github.com/kubernetes-sigs/jobset)? I've added [automatic bootstrapping support for JAX in K8s jobset](https://github.com/jax-ml/jax/pull/22302). This can save the ClusterIP and bootstrapping boilerplates.

Reply:
For example of how to run a jobset, see https://github.com/jax-ml/jax/pull/24197/files

Reply:
Given that we are not currently exposing the code that configures the Kubenetes cluster itself, I was reluctant to go in a direction where the public code (CI config) does not work without non-trivial private logic (JobSet installation, service account configuration, ...)

Also, the bootstrapping doesn't work out of the box because it relies on Python code that is not shipped in our containers. Pursuing that felt like a tangent to the core of this PR.

Reply:
> non-trivial private logic (JobSet installation, service account configuration, ...)

The service account logic is pretty straightforward --- it's not gonna rank in the top N most hairy K8s configs in the world. Plus, we already provided a config that works OOTB!

It's one line to [install jobset](https://github.com/kubernetes-sigs/jobset?tab=readme-ov-file#installation):
```
kubectl apply --server-side -f https://github.com/kubernetes-sigs/jobset/releases/download/v0.7.1/manifests.yaml
```

That said, given that Jobs are 'built-in', maybe we should consider making `jax.distributed.initialize` automatic under it too.

Reply:
> not currently exposing the code that configures the Kubenetes cluster

I wouldn't view the read-only service account and the installation of Jobset as the _sensitive_ information of a cluster. It's the equivalence of one `usermod -aG ...` and one `apt install` for a SLURM cluster.

Reply:
I'm not saying we can't expose it because it's sensitive (it's not). But it's not currently exposed, and it won't be in this PR, so it can't be referred to. It would need to be checked what else is needed re: the AWS/EKS-specific parts of how the roles are managed, too.

> Plus, we already provided a config that works OOTB!

You mean the minikube CI PR to JAX? That's not really discoverable?

The error message still includes a placeholder, BTW: https://github.com/yhtang/jax/blob/2f67710e8ce9cd6c872f85e57417aa9bba7aa270/jax/_src/clusters/k8s_cluster.py#L70

Reply:
> You mean the minikube CI PR to JAX? That's not really discoverable?

I meant my [k8s CI PR](https://github.com/jax-ml/jax/pull/24197/files#diff-3d163a134c02c96a805a5b71ff18a506bc164680cfe152f3806525a64fdeb28f) to upstream.

Reply:
> The error message still includes a placeholder, BTW

You are right... the plan is to upstream another docs-only PR once the CI PR merges.